Tools/Temp/ChangeEditingContextTool.cs

```diff
@@ -0,0 +1,85 @@
+﻿#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using UnityEditor.Experimental.EditorVR.Core;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+using UnityEngine.InputNew;
+
+namespace UnityEditor.Experimental.EditorVR.Tools
+{
+	[MainMenuItem("Change Context", "Create", "Push and pop editing contexts")]
+	//[MainMenuItem(false)]
+	sealed class ChangeEditingContextTool : MonoBehaviour, ITool, ICustomActionMap, ISetEditingContext, ITooltip, 
+		ISetTooltipVisibility, ITooltipPlacement, IUsesRayOrigin
+	{
+		[SerializeField]
```

Comment:
Probably should work with the stack rather than the allAvailable in the manager.  Probably should be populated by an enumeration of available IUnitySceneEditorSubContextLaunchers.  Which in turn should provide a "name" and "icon."

Also, PopContext should probably be moved to its own tool, which is always active.  Such that you can always pop out with the reserved menu button.